test/fw/ptl/lib/pbs_testlib.py

```diff
@@ -13451,6 +13478,61 @@ def action(self, act, script):
         """
         pass
 
+    def add_mom_dyn_res(self, custom_resource, script_body=None,
+                        res_file=None, dirname=None, host=None, perm=0700,
+                        prefix='PtlPbsMomDynRes', suffix='.scr'):
+        """
+        Add a mom dynamic resource script or file to the mom
+        configuration
+
+        :param custom_resource: The name of the custom resource to
+                                define
+        :type custom_resource: str
+        :param script_body: The body of the mom dynamic resource
+        :param res_file: Alternatively to passing the script body, use
+                     the file instead
+        :type res_file: str or None
+        :param dirname: the file will be created in this directory
+        :type dirname: str or None
+        :param host: the hostname on which dyn res script is created
+        :type host: str or None
+        :param perm: perm to use while creating scripts
+                     (must be octal like 0777)
+        :param prefix: the file name will begin with this prefix
+        :type prefix: str
+        :param suffix: the file name will end with this suffix
+        :type suffix: str
+        """
+        if res_file is not None:
+            with open(res_file) as f:
+                script_body = f.readlines()
+                self.du.chmod(host, path=res_file, mode=perm,
+                              sudo=True)
+        else:
+            if dirname is None:
+                dirname = self.pbs_conf['PBS_HOME']
+            tmp_file = self.du.create_temp_file(prefix=prefix, suffix=suffix,
+                                                body=script_body,
+                                                hostname=host)
+
+            res_file = os.path.join(dirname, tmp_file.split(os.path.sep)[-1])
+            self.du.run_copy(host, tmp_file, res_file, sudo=True,
+                             preserve_permission=False)
+            self.du.chown(hostname=host, path=res_file, uid=0, gid=0,
```

Comment:
same comment here, can you please mention in the docstring that the file created will be root owned.